### PR TITLE
[ Fabric ] Adding the support of upgarding chaincode for Fabric 2.2

### PR DESCRIPTION
**Pull request by lakshyakumar**
Signed-off-by: lakshyakumar <lakshya.kumar@accenture.com>

**Changelog**
- Add the support of upgarding chaincode for Fabric 2.2 using `chaincode-install-instantiate.yaml` playbook
- Fix the comments in the playbooks.
- Update operational guide for upgarding the chaincode in Fabric.

 

**Reviewed by**
@jagpreetsinghsasan 

 

**Linked issue**
#1133

### CODE REVIEW

**Review by jagpreetsinghsasan — APPROVED** [review]

File: platforms/hyperledger-fabric/configuration/chaincode-upgrade.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This playbook is a subsequent playbook
 # This playbook does following
-# 1. Upgrades the chaincode provided in network.yaml from each peer
+# 1. Upgrades the chaincode provided in network.yaml from each peer (for fabric version less than 2.2.0 )
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/chaincode-upgrade.yaml:3]: rather mention 1.4.x

File: platforms/hyperledger-fabric/configuration/chaincode-install-instantiate.yaml [hunk 2]
```diff
@@ -2,12 +2,15 @@
 # This playbook does following
 # 1. Installs chaincode provided in network.yaml from each peer
 # 2. Instantiates chaincode from the creator participant peer of the channel( For Fabric 1.4)
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/chaincode-install-instantiate.yaml:4]: remove 1.4 from here or make the comments better for 1.4.x and 2.2.x

File: docs/source/operations/upgrading_chaincode.md [hunk 3]
```diff
@@ -48,14 +51,23 @@ network:
 ```
 
 <a name = "run_network"></a>
-## Run playbook
+## Run playbook for Fabric versions less than 2.2.0
```
Comment (sownak) [inline docs/source/operations/upgrading_chaincode.md:54]: 1.4.x not less than 2.2.0

File: docs/source/operations/upgrading_chaincode.md [hunk 4]
```diff
@@ -1,13 +1,15 @@
 <a name = "upgrading-chaincode"></a>
 # Upgrading chaincode in Hyperledger Fabric
 
-- [Pre-requisites](#pre_req)
-- [Modifying configuration file](#create_config_file)
-- [Running playbook to upgrade chaincode to new version in Hyperledger Fabric network](#run_network)
+- [Upgrading chaincode in Hyperledger Fabric](#upgrading-chaincode-in-hyperledger-fabric)
+  - [Pre-requisites](#pre-requisites)
+  - [Modifying configuration file](#modifying-configuration-file)
+  - [Run playbook for Fabric versions less than 2.2.0](#run-playbook-for-fabric-versions-less-than-220)
```
Comment (sownak) [inline docs/source/operations/upgrading_chaincode.md:7]: 1.4.x and 2.2.x (as we wont be supporting 2.0.0 anyway)

**Review by sownak — CHANGES_REQUESTED** [review]

**Review by sownak — APPROVED** [review]

**Pull request merged**